### PR TITLE
Upgrade undertow and suppress startup messages

### PR DESCRIPTION
**Pull request by jodersky**
Commits are independent. Here' the gist:

---

- Undertow's cookie handling changed slightly in this version, which
  prompted two changes:

  1. the SameSite cookie no longer accepts null as an argument. Rather
     than adding conditionals to guard setting, we simply change the
     default SameSite value to 'Lax', as is assumed by modern standards
    anyway https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie/SameSite:

     > The cookie-sending behavior if SameSite is not specified is SameSite=Lax.
     > Previously the default was that cookies were sent for all requests.
 
  2. it no longer handles spaces in cookie values (unless they are
     quoted). This behavior is actually correct according to
     [RFC2109](https://tools.ietf.org/html/rfc2109), which requires cookie
     values to be quoted if they contain spaces. This change is actually
     transparent to cask, but reveals a minor bug in the
     [requests-scala](https://github.com/lihaoyi/requests-scala/pull/73) library.
     Tests pass locally once this issue is fixed.

- The XNIO framework used by undertow generates a lot of noise when cask
  starts up. These changes introduce a verbosity flag (overridable by the user),
  which slence this noise by default.

  In other words, these changes remove this message from being displayed
  every time cask starts up:

  ```
  Dec 26, 2020 2:37:02 PM io.undertow.Undertow start
  INFO: starting server: Undertow - 2.2.3.Final
  Dec 26, 2020 2:37:02 PM org.xnio.Xnio <clinit>
  INFO: XNIO version 3.8.0.Final
  Dec 26, 2020 2:37:02 PM org.xnio.nio.NioXnio <clinit>
  INFO: XNIO NIO Implementation Version 3.8.0.Final
  Dec 26, 2020 2:37:02 PM org.jboss.threads.Version <clinit>
  INFO: JBoss Threads version 3.1.0.Final
  ```

### CODE REVIEW

**Comment by ManuelWeiss** [comment]
@lihaoyi Can you please merge this PR as version 2.0.13 of undertow has [some critical security issues](https://snyk.io/vuln/SNYK-JAVA-IOUNDERTOW-1012559)? 🙇‍♂️

**Comment by ManuelWeiss** [comment]
🎉 🚀
